libcontainerd/local/local_windows.go

```diff
@@ -435,32 +438,13 @@ func (c *client) Start(_ context.Context, id, _ string, withStdin bool, attachSt
 
 	createProcessParms.User = ctr.ociSpec.Process.User.Username
 
-	ctr.Lock()
-
 	// Start the command running in the container.
 	newProcess, err := ctr.hcsContainer.CreateProcess(createProcessParms)
 	if err != nil {
 		logger.WithError(err).Error("CreateProcess() failed")
-		// Fix for https://github.com/moby/moby/issues/38719.
-		// If the init process failed to launch, we still need to reap the
-		// container to avoid leaking it.
-		//
-		// Note we use the explicit exit code of 127 which is the
-		// Linux shell equivalent of "command not found". Windows cannot
-		// know ahead of time whether or not the command exists, especially
-		// in the case of Hyper-V containers.
-		ctr.Unlock()
-		exitedAt := time.Now()
-		p := &process{
-			id:  libcontainerdtypes.InitProcessName,
-			pid: 0,
-		}
-		c.reapContainer(ctr, p, 127, exitedAt, nil, logger)
```

Comment:
Note for reviewers: containers will not be leaked if creating the init process fails, despite the code being removed here. The daemon handles `Start` errors by cleaning up the container with `(types.Container).Delete`, and the local_windows implementation shuts down and closes the container, same as `reapContainer` did.